### PR TITLE
Datasources: Add user_agent header customization for outgoing HTTP requests

### LINKED ISSUE #63767: Feature request: Enable customizable User-Agent header in outgoing data requests
Issue description is in discussion [Enable customizable User-Agent header in outgoing data requests (e.g. by make BuildVersion customizable in ini file) #58714](https://github.com/grafana/grafana/discussions/58714#discussioncomment-4351906)

### PR DESCRIPTION
**Pull request by zhichli**
Fix #63767 based on [comment](https://github.com/grafana/grafana/discussions/58714#discussioncomment-4351906) in the original discussion #58714

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 1]
```diff
@@ -484,6 +484,10 @@ Limits the amount of bytes that will be read/accepted from responses of outgoing
 
 Limits the number of rows that Grafana will process from SQL (relational) data sources. Default is `1000000`.
 
+### user_agent
+
+Sets a custom value for `User-Agent` header for outgoing data proxy requests. If empty, default value will be `Grafana/<BuildVersion>` (e.g. `Grafana/9.0.0`).
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:489]:
```suggestion
Sets a custom value for the `User-Agent` header for outgoing data proxy requests. If empty, the default value is `Grafana/<BuildVersion>` (for example `Grafana/9.0.0`).
```

**Review by chri2547 — COMMENTED** [review]
Added a suggestion for you. Thank you for the contribution!

File: docs/sources/setup-grafana/configure-grafana/_index.md (hunk 1 above)
Comment (zhichli) [reply docs/sources/setup-grafana/configure-grafana/_index.md]: Thanks for the suggestion! Addressed.

File: pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go [hunk 2]
```diff
@@ -243,7 +242,6 @@ func (e *AzureResourceGraphDatasource) createRequest(ctx context.Context, logger
 	}
 	req.URL.Path = "/"
 	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("User-Agent", fmt.Sprintf("Grafana/%s", setting.BuildVersion))
```
Comment (marefr) [inline pkg/tsdb/azuremonitor/resourcegraph/azure-resource-graph-datasource.go:246]: You seem to have a failed test due to this change

**Review by marefr — CHANGES_REQUESTED** [review]
Thanks. Looks great. A test failure needs to be fixed

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana

**Pull request merged**